packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraSelectorHostApiImpl.java

```diff
@@ -0,0 +1,65 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camerax;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.camera.core.CameraInfo;
+import androidx.camera.core.CameraSelector;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.CameraSelectorHostApi;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CameraSelectorHostApiImpl implements CameraSelectorHostApi {
+  private final BinaryMessenger binaryMessenger;
+  private final InstanceManager instanceManager;
+
+  @VisibleForTesting
+  public CameraSelector.Builder cameraSelectorBuilder = new CameraSelector.Builder();
+
+  public CameraSelectorHostApiImpl(
+      BinaryMessenger binaryMessenger, InstanceManager instanceManager) {
+    this.binaryMessenger = binaryMessenger;
+    this.instanceManager = instanceManager;
+  }
+
+  @Override
+  public Long requireLensFacing(@NonNull Long lensDirection) {
```

Comment:
I think a better pattern would be to change this to a `create` method with this value as a parameter. And then the Dart class would have an additional normal constructor that sets this field. e.g.

```dart
class CameraSelector extends JavaObject {
  CameraSelector({
    int lensFacing,
    super.binaryMessenger,
    super.instanceManager,
  })  : _api = CameraSelectorHostApiImpl(
          binaryMessenger: binaryMessenger,
          instanceManager: instanceManager,
        ),
        super.detached() {
    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
    _api.createFromInstances(this, lensFacing);
  }
}
```

Reply:
And the `create` method 

1. Create the builder
2. set the lens facing.
3.  Build the selector
4. add the selector to the instancemanager

Reply:
Is this the pattern you've typically followed for the builder pattern? I think this makes sense but just wondering about the context on moving away from the Android Api.

Reply:
So in most situations, we do want to follow the Java API because it makes our wrapper API more resistant to changes of the Java API. However, the pattern explained above is often considered, unofficially, as the "Dart equivalent" to the Java [Builder Pattern](https://en.wikipedia.org/wiki/Builder_pattern). The main reason you never see the builder pattern in Dart is because Dart has language features that makes it obsolete.

So in situations like this, where a superior Dart equivalent pattern exists, we can go with the "Dartier" pattern. It should also probably be considered case by case. As far as I can tell, I don't see any downsides to doing it for `CameraSelector`.